qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -26,6 +26,135 @@
 from qiskit_experiments.test.utils import FakeJob
 
 
+class MockRestlessBackend(FakeOpenPulse2Q):
+    """An abstract backend for testing that can mock restless data."""
+
+    def __init__(
+        self,
+        rng_seed: int = 0,
+    ):
+        """
+        Initialize the backend.
+        """
+        self._rng = np.random.default_rng(rng_seed)
+        self._precomputed_probabilities = None
+        super().__init__()
+
+    def _default_options(self):
+        """Default options of the test backend."""
+        return Options(
+            shots=1024,
+            meas_level=MeasLevel.CLASSIFIED,
+            meas_return="single",
+        )
+
+    @abstractmethod
+    def _compute_outcome_probabilities(self,  circuits: List[QuantumCircuit]):
+        """Compute the probabilities of measuring 0 or 1 for each of the given
+         circuits based on the previous measurement shot.
+
+        This methods computes the dictionary self._precomputed_probabilities where
+        the keys are a tuple consisting of the circuit index and the previous outcome,
+        e.g. "0" or "1" for a single qubit. The values are the corresponding probabilities.
+
+        Args:
+            circuits: The circuits from which to compute the probabilities.
+        """
+
+    def run(self, run_input, **options):
+        """Run the restless backend."""
+
+        self.options.update_options(**options)
+        shots = self.options.get("shots")
+        meas_level = self.options.get("meas_level")
+
+        result = {
+            "backend_name": f"{self.__class__.__name__}",
+            "backend_version": "0",
+            "qobj_id": 0,
+            "job_id": 0,
+            "success": True,
+            "results": [],
+        }
+
+        self._compute_outcome_probabilities(run_input)
+
+        prev_outcome = "0"
+        state_strings = ["0", "1"]
```

Comment:
The backend inherits from `FakeOpenPulse2Q` backend which is a two-qubit backend yet these strings enforce a single-qubit setting which is a bit contradictory. Can we change this to
```suggestion
        num_qubits = run_input[0].num_qubits 

        prev_outcome = "0" * num_qubits
        state_strings = self._get_state_strings(num_qubits)
```
Here, I am assuming that `run_input` is a list of circuits. Is that the case? Perhaps a better name would help, e.g. `circuits`?

Reply:
Yes, it is a list of circuits. I used the name `run_input` to make it consistent with the name in the `MockIQBackend`.